### PR TITLE
Added monitoring.coreos.com/v1 crds

### LINKED ISSUE #173: Add monitoring.coreos.com/v1 crds to alloy
### Background

As I was checking the alloy and opportunity to move to alloy instead of grafana-agent, I found out that the crds for PodMonitor, ServiceMonitor, Probe, PodLogs are not present in alloy
In the documents for Migrating grafana agent operator to alloy there is a part which states: "The Monitor types (PodMonitor, ServiceMonitor, Probe, and PodLogs) are all supported natively by Alloy."
but without the crds, I think it is pointless

### Proposal

I propose that we should add these crds and there should be a flag to enable to install them within values.yaml

### PR DESCRIPTION
**Pull request by itspooya**
#### PR Description

This adds monitoring.coreos.com/v1 crds, more info in issue

#### Which issue(s) this PR fixes

Fixes #173

#### PR Checklist

- [ ] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

**Comment by rfratto** [comment]
Hey! This is something we've been back and forth on a few times, but I think it's reasonable to start including the CRDs from Prometheus Operator.

I'm worried that copying/pasting them risks us getting out of sync fairly quickly. What would it look like if we added a chart dependency on the [official Chart](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-operator-crds) which contains the CRDs? (Even though not all of the CRDs are support in Alloy).

This way, we can allow other tooling to keep us up to date.

**Comment by itspooya** [comment]
That's a good idea, I saw that before but the question is, do we need all the crds?
what are your suggestions on this?

**Comment by rfratto** [comment]
Given the trade-off between being in sync with upstream and installing some unnecessary CRDs, I'd generally prefer being in sync right now, especially since CRDs are pretty low overheard (IIUC). If this is a problem in the future we can consider rolling back the set of CRDs to just the set we need and figuring out a strategy to not fall out of sync.

**Comment by itspooya** [comment]
Then based on your answer, I assume I should make a subchart that installs these crds, is that right?
if it is the case, I will make the changes and then ask you again to review
Thanks

**Comment by rfratto** [comment]
Yes please :)

**Comment by itspooya** [comment]
I will do the changes and let you know
I am excited about this, thanks for the opportunity

File: operations/helm/charts/alloy/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ This document contains a historical list of changes between releases. Only
 changes that impact end-user behavior are listed; changes to documentation or
 internal API changes are not present.
 
+0.2.0 (2024-04-??)
+------------------
+
+- Add prometheus-operator-crds as a dependency
+
```
Comment (tpaschalis) [inline operations/helm/charts/alloy/CHANGELOG.md:14]: Hey @itspooya thanks a lot for the PR! I think this makes sense and I'll leave the fial word to @\rfratto but adding two small comments

We're actually not releasing a new Helm version for _every_ change but prefer to batch them and use our tooling to release on demand. 

```suggestion
Unreleased
----------

- Add prometheus-operator-crds as a dependency

```

File: operations/helm/charts/alloy/README.md [hunk 2]
```diff
@@ -114,6 +114,7 @@ useful if just using the default DaemonSet isn't sufficient.
 | ingress.pathType | string | `"Prefix"` |  |
 | ingress.tls | list | `[]` |  |
 | nameOverride | string | `nil` | Overrides the chart's name. Used to change the infix in the resource names. |
+| prometheus-crds.create | bool | `true` | Whether to install CRDs for Prometheus. |
```
[the remark this answers is not in the document]
Comment (itspooya) [reply operations/helm/charts/alloy/README.md]: I have used helm-docs and updated the documents

**Comment by itspooya** [comment]
I have added a line into test part, which would run helm dependency update

**Comment by itspooya** [comment]
Hi, about the failing CIs, what should I do?

**Pull request closed without merging**